### PR TITLE
Ft/es orders

### PR DESCRIPTION
**Pull request by Araknyfe**
# Description

New ES service, get and persist functions for orders, possibilty to add event.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Traineau — APPROVED** [review]

**Comment by Araknyfe** [comment]
Sorry, I had missed that this PR had not been merged.

File: router/router.go [hunk 1]
```diff
@@ -108,6 +108,13 @@ var routes = Routes{
 		HandlerFunc: auth.Refresh,
 		Public:      false,
 	},
+	Route{
+		Name:        "Create an order",
+		Method:      "POST",
+		Pattern:     "/orders",
+		HandlerFunc: controllers.CreateOrder,
+		Public:      true, //@TODO : switch to false, testing purposes
```
Comment (AlexandreLch) [inline router/router.go]: remove TODO and set to false

**Review by AlexandreLch — APPROVED** [review]

File: router/router.go [hunk 2]
```diff
@@ -108,6 +108,13 @@ var routes = Routes{
 		HandlerFunc: auth.Refresh,
 		Public:      false,
 	},
+	Route{
```
Comment (Traineau) [inline router/router.go]: On a deja cette route : https://github.com/HETIC-MT-P2021/CQRSES_GROUP5/blob/ca18aada25d2e244d05f8a89c2b45b0ffee2e497/router/router.go#L55

File: models/order.go [hunk 3]
```diff
@@ -4,7 +4,7 @@ import "time"
 
 //Order is the basic model for order ( which are sent in commands )
 type Order struct {
-	ID         uint         `json:"id"`
+	ID         string       `json:"id"`
```
Comment (Traineau) [inline models/order.go:7]: Tous les ID sont des string maintenant ? T'as vérifier que toutes les conversions passent bien pour RabbitMQ ?

File: database/es_connection.go [hunk 4]
```diff
@@ -19,11 +19,12 @@ type ConfigEs struct {
 
 //ConnectES creates a new ES client and stores it
 func ConnectES(ctx context.Context, cfg *ConfigEs, foreverLoopDelay time.Duration) error {
+
 	client, err := elastic.NewClient(
 		elastic.SetHealthcheck(true),
 		elastic.SetSniff(true),
 		elastic.SetURL(cfg.URL),
-		elastic.SetHealthcheckInterval(15*time.Second),
+		elastic.SetHealthcheckInterval(100*time.Second),
```
Comment (Traineau) [inline database/es_connection.go]: 100 ? ça va etre long a up le docker

File: router/router.go (hunk 2 above)
Comment (Araknyfe) [reply router/router.go]: Yep je viens de l'enlever sorry c'etait pour mes tests

File: database/es_connection.go (hunk 4 above)
Comment (Araknyfe) [reply database/es_connection.go]: ah oui tu avais fix ça je crois de ton coté, désolée, j'avais laissé la PR en draft j'étais persuadée qu'elle etait merge

File: router/router.go (hunk 1 above)
Comment (Araknyfe) [reply router/router.go]: Done

**Review by Traineau — APPROVED** [review]

**Pull request merged**